src/DotVVM.Framework/Compilation/Binding/ExpressionHelper.cs

```diff
@@ -147,36 +155,74 @@ public static Expression Call(Expression target, Expression[] arguments)
+            return newArguments;
+        }
+
         public static Expression CallMethod(Type target, BindingFlags flags, string name, Type[] typeArguments, Expression[] arguments, IDictionary<string, Expression> namedArgs = null)
         {
             // the following piece of code is nicer and more readable than method recognition done in roslyn, C# dynamic and also expression evaluator :)
-            var method = FindValidMethodOveloads(target, name, flags, typeArguments, arguments, namedArgs);
+            var method = FindValidMethodOveloads(null, target, name, flags, typeArguments, arguments, namedArgs);
             return Expression.Call(method.Method, method.Arguments);
         }
 
 
-        private static MethodRecognitionResult FindValidMethodOveloads(Type type, string name, BindingFlags flags, Type[] typeArguments, Expression[] arguments, IDictionary<string, Expression> namedArgs)
+        private static MethodRecognitionResult FindValidMethodOveloads(Expression target, Type type, string name, BindingFlags flags, Type[] typeArguments, Expression[] arguments, IDictionary<string, Expression> namedArgs)
         {
-            var methods = FindValidMethodOveloads(type.GetAllMembers(flags).OfType<MethodInfo>().Where(m => m.Name == name), typeArguments, arguments, namedArgs).ToList();
+            var methods = FindValidMethodOveloads(type.GetAllMembers(flags).OfType<MethodInfo>().Where(m => m.Name == name), typeArguments, arguments, namedArgs).ToList();           
 
             if (methods.Count == 1) return methods.FirstOrDefault();
-            if (methods.Count == 0) throw new InvalidOperationException($"Could not find overload of method '{name}'.");
-            else
+            if (methods.Count == 0)
             {
-                methods = methods.OrderBy(s => s.CastCount).ThenBy(s => s.AutomaticTypeArgCount).ToList();
-                var method = methods.FirstOrDefault();
-                var method2 = methods.Skip(1).FirstOrDefault();
-                if (method.AutomaticTypeArgCount == method2.AutomaticTypeArgCount && method.CastCount == method2.CastCount)
+                // We did not find any match in regular methods => try extension methods
+                if (target != null)
                 {
-                    // TODO: this behavior is not completed. Implement the same behavior as in roslyn.
-                    throw new InvalidOperationException($"Found ambiguous overloads of method '{name}'.");
+                    // Change to a static call
+                    var newArguments = CreateArgumentsForExtensionMethod(target, arguments);
+                    var extensions = FindValidMethodOveloads(type.GetAllExtensions().OfType<MethodInfo>().Where(m => m.Name == name), typeArguments, newArguments, namedArgs).ToList();
+
+                    // We found an extension method
+                    if (extensions.Count == 1)
+                    {
+                        extensions[0].IsExtension = true;
```

Comment:
Shall we set `IsExtension = true` for all found extension methods?

Reply:
Yes - thanks I did not notice that :). Fixed